cores/esp8266/AddrList.h

```diff
@@ -124,7 +124,7 @@ struct netifWrapper
 
     // common to all addresses of this interface
     String ifname () const          { return String(_netif->name[0]) + _netif->name[1]; }
-    const char* ifhostname () const { return _netif->hostname?: emptyString.c_str(); }
```

Comment:
[It is valid and returns the tested value if not '0'](https://gcc.gnu.org/onlinedocs/gcc/Conditionals.html#Conditionals)